### PR TITLE
[Clean/SAM] add common functions - @open sesame 09/18 15:27

### PR DESCRIPTION
**Pull request by jaeyun-jung**
1. define common functions to reduce complexity
2. condition to print log message about nnapi

Signed-off-by: Jaeyun Jung <jy1210.jung@samsung.com>

### CODE REVIEW

File: api/capi/include/nnstreamer-capi-private.h [hunk 1]
```diff
@@ -278,6 +284,31 @@ void ml_tensors_info_copy_from_ml (GstTensorsInfo *gst_info, const ml_tensors_in
  */
 GstCaps * ml_tensors_info_get_caps (const ml_tensors_info_s *info);
 
+/**
+ * @brief Creates a tensor data frame wihout buffer with the given tensors information.
+ * @param[in] info The handle of tensors information for the allocation.
+ * @param[out] data The handle of tensors data.
+ * @return @c 0 on success. Otherwise a negative error value.
```
Comment (leemgs) [inline api/capi/include/nnstreamer-capi-private.h]: I have a minor question. What is the meaning of @c?

File: api/capi/src/nnstreamer-capi-util.c [hunk 2]
```diff
@@ -806,6 +806,104 @@ ml_tensors_info_get_caps (const ml_tensors_info_s * info)
...
+ * @since_tizen 5.5
+ * @param[in] model The path of model file.
+ * @param[in/out] nnfw The type of NNFW.
+ * @return @c 0 on success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Given parameter is invalid.
+ */
+int
+ml_validate_model_file (const char *model, ml_nnfw_type_e * nnfw)
```
Comment (leemgs) [inline api/capi/src/nnstreamer-capi-util.c]: Looks good.

File: api/capi/src/nnstreamer-capi-single.c [hunk 3]
```diff
@@ -119,6 +118,78 @@ ml_single_get_tensors_info_from_filter (GstElement * filter, gboolean is_input,
...
+  g_free (str_dim);
+  g_free (str_type);
+  g_free (str_name);
+  gst_tensors_info_free (&gst_info);
+
+  return option;
+}
+
+/**
+ * @brief Internal static function to set tensors info in handle.
```
Comment (leemgs) [inline api/capi/src/nnstreamer-capi-single.c]: Internal static function to set tensors info in **the** handle

File: api/capi/include/nnstreamer-capi-private.h (hunk 1 above)
Comment (jaeyun-jung) [reply api/capi/include/nnstreamer-capi-private.h]: display font
```
@a italics
@c typewriter font
```

File: api/capi/src/nnstreamer-capi-single.c (hunk 3 above)
Comment (jaeyun-jung) [reply api/capi/src/nnstreamer-capi-single.c]: updated

**Pull request merged**